packages/core/contracts/financial-templates/common/financial-product-libraries/KpiOptionsFinancialProductLibrary.sol

```diff
@@ -11,6 +11,42 @@ import "../../../common/implementation/Lockable.sol";
  * Post-expiry, the collateral requirement is left as 1 and the price is left unchanged.
  */
 contract KpiOptionsFinancialProductLibrary is FinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    mapping(address => FixedPoint.Unsigned) financialProductTransformedPrices;
+
+    /**
+     * @notice Enables any address to set the transformed price for an associated financial product.
+     * @param financialProduct address of the financial product.
+     * @param transformedPrice the price for the financial product to be used if the contract is pre-expiration.
+     * @dev Note: a) Any address can set price transformations b) The price can't be set to blank.
+     * c) A transformed price can only be set once to prevent the deployer from changing it after the fact.
+     * d)  financialProduct must expose an expirationTimestamp method.
+     */
+    function setFinancialProductTransformedPrice(address financialProduct, FixedPoint.Unsigned memory transformedPrice)
+        public
+        nonReentrant()
+    {
+        require(transformedPrice.isGreaterThan(0), "Cant set price of 0");
```

Comment:
```suggestion
        require(transformedPrice.isGreaterThan(0), "Cannot set price to 0");
```

Reply:
Also, are we ok not allowing 0 prices? For example, this library would be bad for the funding rate price identifier whose value could realistically be 0.

Reply:
the original idea was it would prevent you from setting uninitialised prices but that is a good point @nicholaspai

Reply:
I refactored this a bit to allow a price of 0, for cases where that is useful.